server/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java

```diff
@@ -329,6 +334,7 @@ public void apply(Settings value, Settings current, Settings previous) {
         ProxyConnectionStrategy.SERVER_NAME,
         SniffConnectionStrategy.REMOTE_CLUSTERS_PROXY,
         SniffConnectionStrategy.REMOTE_CLUSTER_SEEDS,
+        SniffConnectionStrategy.REMOTE_CLUSTER_AUTHORIZATION,
```

Comment:
```suggestion
        ClusterSettings.CCX2_FEATURE_FLAG_ENABLED ? SniffConnectionStrategy.REMOTE_CLUSTER_AUTHORIZATION : null,
```

See my suggestion below. This way we keep the logic with the setting and can rely on the existing null filter.

Reply:
I applied the change.